### PR TITLE
8319986: Invalid/inconsistent description and example for DateFormat

### PR DESCRIPTION
**Pull request by naotoj**
Correcting the explanation of the `DateFormat.SHORT` constant.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8319986](https://bugs.openjdk.org/browse/JDK-8319986): Invalid/inconsistent description and example for DateFormat (**Bug** - P3)

### Reviewers
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16645/head:pull/16645` \
`$ git checkout pull/16645`

Update a local copy of the PR: \
`$ git checkout pull/16645` \
`$ git pull https://git.openjdk.org/jdk.git pull/16645/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16645`

View PR using the GUI difftool: \
`$ git pr show -t 16645`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16645.diff">https://git.openjdk.org/jdk/pull/16645.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16645#issuecomment-1809307361)

### CODE REVIEW

File: src/java.base/share/classes/java/text/DateFormat.java [hunk 1]
```diff
@@ -117,7 +117,7 @@
  * different options to these factory methods to control the length of the
  * result; from {@link #SHORT} to {@link #MEDIUM} to {@link #LONG} to {@link #FULL}. The exact result depends
  * on the locale, but generally:
- * <ul><li>{@link #SHORT} is completely numeric, such as {@code 12.13.52} or {@code 3:30pm}
+ * <ul><li>{@link #SHORT} is the shortest and mainly numeric, such as {@code 12.13.52} or {@code 3:30pm}
```
Comment (JoeWang-Java) [inline src/java.base/share/classes/java/text/DateFormat.java:120]: Not sure if we want to update the descriptions of all options, but if we want to be consistent, here's a suggestion, how about:
  {@link #SHORT} is a concise format using numeric digits, for example, {@code 12.13.52} or {@code 3:30pm}
  {@link #MEDIUM} provides more detail, such as {@code Jan 12, 1952}
  {@link #LONG} is a comprehensive format, such as {@code January 12, 1952} or {@code 3:30:32pm}
  {@link #FULL} provides a complete date representation, such as
  
  "pretty completely" is a bit casual :-)

Comment (naotoj) [reply src/java.base/share/classes/java/text/DateFormat.java:120]: Thanks for the feedback, Joe. I think I would rather keep the description somewhat abstract, simply mentioning the formatted output is `SHORT` <= `MEDIUM` <= `LONG` <= `FULL` in length-wise. As it is noted in the previous sentence, it is pretty much locale-dependent.
I simply replaced that casual wording with "the longest" for consistency.

File: src/java.base/share/classes/java/text/DateFormat.java [hunk 2]
```diff
@@ -117,10 +117,10 @@
...
  * <li>{@link #MEDIUM} is longer, such as {@code Jan 12, 1952}
  * <li>{@link #LONG} is longer, such as {@code January 12, 1952} or {@code 3:30:32pm}
```
Comment (justin-curtis-lu) [inline src/java.base/share/classes/java/text/DateFormat.java:122]: Does it make sense to change LONG to "_is even longer_" or "_is longer than MEDIUM_"?

Still keeps it abstract but clarifies it a little more IMO (and should still hold true regardless of locale)

Comment (naotoj) [reply src/java.base/share/classes/java/text/DateFormat.java]: Good point. Replaced as suggested.

File: src/java.base/share/classes/java/text/DateFormat.java (hunk 1 above)
Comment (naotoj) [reply src/java.base/share/classes/java/text/DateFormat.java:120]: Added a sentence explaining it.

**Review by JoeWang-Java — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Review by justin-curtis-lu — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Pull request closed without merging**